### PR TITLE
doc/fix: Add ocaml 5.1.1 as melange v3 dependency

### PR DESCRIPTION
**Pull request by sravan-s**
Beginner here, this gave me some trouble installing melange following -
https://github.com/melange-re/melange?tab=readme-ov-file#installation

Might save 5 minutes for someone in the future

https://github.com/melange-re/melange/pull/926

### CODE REVIEW

**Comment by anmonteiro** [comment]
This is not true, though. We released versions for 4.14 too

**Comment by sravan-s** [comment]
Okok, I didnt know this. Anyways, take / add the info if you think the information is useful :)

File: README.md [hunk 1]
```diff
@@ -66,6 +66,7 @@ Melange has good support for Nix:
 
 ### OCaml version compatibility
 
+- Melange v3.0 works on OCaml (>=4.14 & <5.0) or 5.1.1.
```
Comment (anmonteiro) [inline README.md:69]:
```suggestion
- Melange v3.0 works on OCaml 4.14 ([v3-414](https://github.com/melange-re/melange/releases/tag/3.0.0-414)) and 5.1.1 ([v3-51](https://github.com/melange-re/melange/releases/tag/3.0.0-51)).
```

File: README.md [hunk 2]
```diff
@@ -66,7 +66,7 @@ Melange has good support for Nix:
 
 ### OCaml version compatibility
 
-- Melange v3.0 works on OCaml (>=4.14 & <5.0) or 5.1.1.
+- Melange v3.0 works on OCaml 4.14 ([v3-414](https://github.com/melange-re/melange/releases/tag/3.0.0-414)) and 5.1 ([v3-51](https://github.com/melange-re/melange/releases/tag/3.0.0-51)).
```
Comment (sravan-s) [inline README.md:69]: :+1:

**Comment by anmonteiro** [comment]
Thansk

**Pull request merged**